File: En Language Python/Lineaire.py
#Fonction pour effectuer une recherche linéaire

def recherche_lineaire(tableau, element):
    for i in range(len(tableau)):
        if tableau[i] == element:
            return i  # Élément trouvé, retourner l'indice
    return -1  # Élément non trouvé


# Exemple d'utilisation
tableau = [2, 3, 4, 10, 40]
element = int(input("Entrez l'élément à rechercher : "))
resultat = recherche_lineaire(tableau, element)

if resultat != -1:
    print(f"Élément trouvé à l'index {resultat}")
else:
    print("Élément non trouvé dans le tableau")

File: En Language C/Lineaire.c
#include <stdio.h>

// Fonction pour effectuer une recherche linéaire
int rechercheLineaire(int tableau[], int taille, int element) {
    for (int i = 0; i < taille; i++) {
        if (tableau[i] == element) {
            return i; 
        }
    }
    return -1;
}

int main() {
    int tableau[] = {2, 3, 4, 10, 40};
    int taille = sizeof(tableau) / sizeof(tableau[0]);
    int element = 10;
    int resultat = rechercheLineaire(tableau, taille, element);

    if (resultat != -1) {
        printf("Élément trouvé à l'index %d\n", resultat);
    } else {
        printf("Élément non trouvé dans le tableau\n");
    }

    return 0;
}
